### ISSUE TITLE
pushover and pushbullet does not work

### ISSUE THREAD

**Issue by fdvalero** [sriccio/zabbix-alertscripts#3]

**Comment by sriccio** [comment]
Hi,

Your issue lacks a bit of information. Did you try to run the scripts from a shell before calling it from Zabbix.

Does it return an error ?

Did you installed python dependencies?

**Comment by fdvalero** [comment]
I tried:
python pushover.py userkey|apptoken test

And does not work, I ran this in too many ways.

`python pushover.py [censored]-[censored] "testing"
usage: pushover.py [-h] UserKey|AppToken Subject Message
pushover.py: error: too few arguments


python pushover.py censored|censored"testing"
bash: censored: command not found...
usage: pushover.py [-h] UserKey|AppToken Subject Message
pushover.py: error: too few arguments

[root@antlet40 alertscripts]# python pushover.py -h censored|censoredtest
bash: censored: command not found...
close failed in file object destructor:
sys.excepthook is missing
lost sys.stderr
[root@antlet40 alertscripts]# python pushover.py -h "censored|censoredtest
usage: pushover.py [-h] UserKey|AppToken Subject Message

Send Zabbix notification to Pushover enabled devices.

positional arguments:
  UserKey|AppToken  Pushover User Key AND Application Token separated by |
  Subject           Subject you want to push to the device(s).
  Message           Message you want to push to the device(s).

optional arguments:
  -h, --help        show this help message and exit
[root@antlet40 alertscripts]# python pushover.py -h 'censored|censored' test
usage: pushover.py [-h] UserKey|AppToken Subject Message

Send Zabbix notification to Pushover enabled devices.

positional arguments:
  UserKey|AppToken  Pushover User Key AND Application Token separated by |
  Subject           Subject you want to push to the device(s).
  Message           Message you want to push to the device(s).

optional arguments:
  -h, --help        show this help message and exit
[root@antlet40 alertscripts]# 


`

Can you please tell me what is the way to test this?

Regards

**Comment by fdvalero** [comment]
I will want to make this work, please help me.
If you want I can send you my telegram or whatapp contact information.

regards
Fidel Valero

**Comment by sriccio** [comment]
try:

`python pushover.py "censored|censored" "test subject" "test message"`

In some of your examples you keep repeating -h argument but that is to display help.
In others you don't use double quotes around the the arguments and then linux will interpret | as pipe.
In some you use a double quote but there is not ending double quote.

**Comment by fdvalero** [comment]
The same
 `[root@antlet40 alertscripts]# python pushover.py -h "censored|censored" "test"
usage: pushover.py [-h] UserKey|AppToken Subject Message

Send Zabbix notification to Pushover enabled devices.

positional arguments:
  UserKey|AppToken  Pushover User Key AND Application Token separated by |
  Subject           Subject you want to push to the device(s).
  Message           Message you want to push to the device(s).

optional arguments:
  -h, --help        show this help message and exit
[root@antlet40 alertscripts]# 
`

I don't know if I need another prerequisites I've installed the python-pushover.

**Comment by fdvalero** [comment]
I just notice in the script, I don't see where the url api.pushover.net/api/..... Is defined

**Comment by sriccio** [comment]
Why do you add -h to the command line ? it's to display help.
In all your tests you add this parameter, so for sure it will always print the help.

**Comment by sriccio** [comment]
The pushover api url is defined in the python-pushover python module which is a dependency to use the script.

**Comment by fdvalero** [comment]
ok, but still not working in zabbix 3.4


Regards

**Comment by sriccio** [comment]
Were you able to send a notification using it from command line ?

If yes, check that you configured the script in zabbix with the correct arguments.

{ALERT.SENDTO}
{ALERT.SUBJECT}
{ALERT.MESSAGE}

Check that the paths to alert scripts are correct and that the user which runs the zabbix-server has permission to access and run scripts in this path.

Check the log file from the script if you enabled it and rise the log level of your zabbix-server to be more verbose about what it tries to do and the result.

Screenshots are from Zabbix 3.2 but it works perfectly fine on our different Zabbix updated to 3.4.

Edit: The most useful thing in your case is to raise the log level of your zabbix server. It will produce a lot of output in the log but then you can search it for "pushbullet" or "pushover" to find calls to the scripts and their results-

**Comment by fdvalero** [comment]
How do you pretend to work in the zabbix if via cli does not work as you see in the output that I send you. I'm not able to see this logs because even the system does not touch the logs.
If you have another documentation steps by steps is ok. But this one does not work as you explain in the readme.

**Comment by sriccio** [comment]
I pretend it works on Zabbix 3.4 because we use it on multiple installs of Zabbix 3.4 every day.

It's the third time I tell you to stop using -h argument on your tests because it's to display help.

All your tests were made with -h. The argument is quite clear or ? 

-h, --help show this help message and exit

**Comment by sriccio** [comment]
Please read again my comment here:

https://github.com/sriccio/zabbix-alertscripts/issues/3#issuecomment-353748136

**Comment by fdvalero** [comment]
[root@antlet40 alertscripts]# python notify_pushover.py
'censored|censored' 'test'
usage: notify_pushover.py [-h] UserKey|AppToken Subject Message
notify_pushover.py: error: too few arguments
[root@antlet40 alertscripts]#
[root@antlet40 alertscripts]#
[root@antlet40 alertscripts]# python notify_pushover.py
"censored|censored" "test"
usage: notify_pushover.py [-h] UserKey|AppToken Subject Message
notify_pushover.py: error: too few arguments
[root@antlet40 alertscripts]#


This this the output.

**Comment by sriccio** [comment]
Your last test is quite right, just missing the "message" argument

The error returned says that you're missing an arugment
`notify_pushover.py: error: too few arguments`

So try:
`python notify_pushover.py "censored|censored" "test subject" "test message"`

**Comment by sriccio** [comment]
But it's true that it's confusing because the argparser should put quotes around "Subject" and "Message"

**Comment by fdvalero** [comment]
ok, and now I have this:

`[root@antlet40 alertscripts]# python notify_pushover.py "censored|censored" "test subject" "test message"
Traceback (most recent call last):
  File "notify_pushover.py", line 98, in <module>
    except (pushover.UserError) as exc:
AttributeError: 'module' object has no attribute 'UserError'
`

**Comment by sriccio** [comment]
Did you install python-pushover package, with pip ?

`pip install python-pushover`

Which python version are you using ?

**Comment by fdvalero** [comment]
`[root@antlet40 alertscripts]# pip install python-pushover
Requirement already satisfied: python-pushover in /usr/lib/python2.7/site-packages
Requirement already satisfied: requests>=1.0 in /usr/lib64/python2.7/site-packages (from python-pushover)
Requirement already satisfied: certifi>=2017.4.17 in /usr/lib/python2.7/site-packages (from requests>=1.0->python-pushover)
Requirement already satisfied: chardet<3.1.0,>=3.0.2 in /usr/lib/python2.7/site-packages (from requests>=1.0->python-pushover)
Requirement already satisfied: idna<2.7,>=2.5 in /usr/lib/python2.7/site-packages (from requests>=1.0->python-pushover)
Requirement already satisfied: urllib3<1.23,>=1.21.1 in /usr/lib/python2.7/site-packages (from requests>=1.0->python-pushover)
`

This is for ppython-pushover


Python:
`[root@antlet40 alertscripts]# rpm -qa | grep ^python
python-pycurl-7.19.0-19.el7.x86_64
python-libs-2.7.5-58.el7.x86_64
python-pyblock-0.53-6.el7.x86_64
python-javapackages-3.4.1-11.el7.noarch
python-backports-ssl_match_hostname-3.4.0.2-4.el7.noarch
python-nss-0.16.0-3.el7.x86_64
python-slip-dbus-0.4.0-2.el7.noarch
python-meh-0.25.2-1.el7.noarch
python-chardet-2.2.1-1.el7_1.noarch
python-configobj-4.7.2-7.el7.noarch
python-inotify-0.9.4-4.el7.noarch
python-deltarpm-3.6-3.el7.x86_64
python-gobject-3.22.0-1.el7.x86_64
python-IPy-0.75-6.el7.noarch
python-firewall-0.4.4.4-6.el7.noarch
python-kitchen-1.1.1-5.el7.noarch
python-schedutils-0.4-6.el7.x86_64
python-perf-3.10.0-693.el7.x86_64
python-coverage-3.6-0.5.b3.el7.x86_64
python2-pip-8.1.2-5.el7.noarch
python-blivet-0.61.15.65-1.el7.noarch
python-urlgrabber-3.10-8.el7.noarch
python-gobject-base-3.22.0-1.el7.x86_64
python-decorator-3.4.0-3.el7.noarch
python-augeas-0.5.0-2.el7.noarch
python-pwquality-1.2.3-4.el7.x86_64
python-lxml-3.2.1-4.el7.x86_64
python-six-1.9.0-2.el7.noarch
python-iniparse-0.4-9.el7.noarch
python-backports-1.0-8.el7.x86_64
python-ethtool-0.8-5.el7.x86_64
python2-caribou-0.4.21-1.el7.noarch
python-meh-gui-0.25.2-1.el7.noarch
python-pyudev-0.15-9.el7.noarch
python-slip-0.4.0-2.el7.noarch
python-linux-procfs-0.4.9-3.el7.noarch
python-di-0.3-2.el7.noarch
python-ntplib-0.3.2-1.el7.noarch
python-brlapi-0.6.0-15.el7.x86_64
python-smbc-1.0.13-7.el7.x86_64
python-2.7.5-58.el7.x86_64
python-dmidecode-3.12.2-1.el7.x86_64
python-cups-1.9.63-6.el7.x86_64
python-setuptools-0.9.8-7.el7.noarch
[root@antlet40 alertscripts]# 
`

**Comment by sriccio** [comment]
Tested on both python2 and python3 with the python-pushover dependency.

Python2:
`root@eye2# python2 notify_pushover.py "censored|censored" "test subject" "test message"              
[Sun Dec 24 02:56:20 2017] Success: Message sent with UserKey [censored] to AppToken [censored]
`

Python3
`
root@eye2# python3 notify_pushover.py "censored|censored" "test subject" "test message"  
[Sun Dec 24 02:56:28 2017] Success: Message sent with UserKey [censored] to AppToken [censored]
`

**Comment by sriccio** [comment]
What is 

`pip list | grep push`

Reporting ?

**Comment by fdvalero** [comment]
`[root@antlet40 alertscripts]# pip list | grep push
DEPRECATION: The default format will switch to columns in the future. You can use --format=(legacy|columns) (or define a format=(legacy|columns) in your pip.conf under the [list] section) to disable this warning.
pushbullet.py (0.11.0)
pushover (0.4)
python-pushover (0.3)
python-pushsafer (0.3)
[root@antlet40 alertscripts]# 
`

**Comment by sriccio** [comment]
You have a wrong pushover package installed...

pushover (0.4) is not the same as python-pushover and python is including the wrong one.

`pip uninstall pushover`

**Comment by fdvalero** [comment]
Good eyes, I spend like 5 hours on this, and finally thanks to you is working:

`[root@antlet40 alertscripts]# python notify_pushover.py "censored|censored" "testing" "blablabla"
[Sat Dec 23 18:06:31 2017] Success: Message sent with UserKey [censored] to AppToken [censored]
[root@antlet40 alertscripts]# 
`

I will see now in zabbix.

**Comment by sriccio** [comment]
Okay :)

**Comment by sriccio** [comment]
Note: I've edited the messages and replaced tokens and ids with censored. In case there were real ID's.

**Comment by fdvalero** [comment]
I tested the pushbullet via cli, and works very well, now the problem is in the configuration in the Zabbix GUI. I still not receive the notification.

**Comment by fdvalero** [comment]
The script need to be owned by zabbix?

**Comment by fdvalero** [comment]
I will re-open this because I don't get the notifications on th phone. I tested it via cli as you showed me, and works. But still not received the alarm in the phone.
I set correctly the script parameter:
{ALERT.SENDTO}
{ALERT.SUBJECT}
{ALERT.MESSAGE}

And the UserKey|AppToken set in Media on the Admin user.

Any idea what else may I check. Also I want to mention that I did not touch the "option" in Media Type.
